### PR TITLE
Add svg maps implementation

### PR DESCRIPTION
**Pull request by jpmonettas**
Implements a bunch of functionality needed to visualize analysis
output data into svg maps using only reagent/re-frame as a dependency.

Client side highlights :
 - Implements a svg-renderer to render geojson maps with reagent
 - Implement a reagent animated map component, capable of rendering data
   at different points in time and animate it.
 - Add a test page for maps under /map, to retrieve a analysis by id
   and render it on a map

Server side highlights :
 - Add all maps encoded as geojson resources
 - Add functionality to upload map files to a S3 bucket
 - Add maps to graphql analysis schema (only continuous tree so far)

### CODE REVIEW

File: resources/schema.edn
Comment (fbielejec) [inline resources/schema.edn:69]: No entiendo, what is this comment referring to? Is this the discussion we had about having a common Analysis Interface  and deriving Continuos / DiscreteTree etc form it?

File: src/clj/api/resolvers.clj
Comment (fbielejec) [inline src/clj/api/resolvers.clj:127]: I have no problem storing the bounding box in the DB, but we also discussed that the SpreadData json file can be self-contained and the bounding box can be obtained by following the AxisAttribute.x / AxisAttribute.y pointers to the Attributes list and reading the range field that they have. ContinuousTree parser definitely has this data and we can make sure that the other analysis parsers store it to. Your call, I can add this information to the other Java parsers in a next PR.

File: src/clj/api/server.clj
Comment (fbielejec) [inline src/clj/api/server.clj:166]: I implemented this once, reading from a jar archive:https://github.com/phylogeography/SpreaD3/blob/master/src/renderers/d3/D3Renderer.java#L36
Maybe helpfull, if not, yeha we can have this one-off manual step in deployment, no kittens should die.

File: src/cljc/shared/math_utils.cljc
Comment (fbielejec) [inline src/cljc/shared/math_utils.cljc]: officially the smartest code I've seen written in Clojure :)

File: src/cljc/shared/math_utils.cljc
Comment (fbielejec) [inline src/cljc/shared/math_utils.cljc]:
```suggestion
  "Calculates the length of a quadratic curve given three points using the algorithm in
```

File: src/cljs/ui/component/maps.cljs
Comment (fbielejec) [inline src/cljs/ui/component/maps.cljs:202]: empty map needed?

File: src/cljs/ui/component/maps.cljs
Comment (fbielejec) [inline src/cljs/ui/component/maps.cljs:212]: formatting pls, these parens are lonely here ;)

File: src/cljs/ui/config.cljs
Comment (fbielejec) [inline src/cljs/ui/config.cljs:22]: I imagined that the map renderer is more of a separate application, that runs it's own process and accepts s3 URL as it's only argument, and renders the SpreadData (json) content? Unless this is for development purposes and we will split it up later?

**Review by fbielejec — COMMENTED** [review]
This is good work! I left a few minor comments, my major one is that I imagined the renderer to be a separate client application (separate cljs build), running in it's own process w-out even calling graphQL, using authorization or the server for that matter.
That also corresponds to the designs: https://xd.adobe.com/view/cab84bb6-15c6-44e3-9458-2ff4af17c238-9feb/screen/55c6be2b-7282-4f99-a74b-17f81a79997e/ 

The share-able URL opens the map in a separate browser tab (a stand-alone SPA), It could accept two arguments:

- the S3 URL pointing to the SpreadData and
- another URL with map background, provided that the queue worker does the work of "matching" the data bounding box to a map and persisting that information. But in that case we need to guarantee that the content of that S3 folder is immutable or at least that the URLs are preserved if we alter the map data. Alternatively, the renderer client application could make one server call to an (un-authorized) endpoint returning the list of maps + their respective bounding boxes and match on the client (O(N) operation, should be doable in most browsers). I think I like the former better (worker matches the map to the data and we store both the output Spread data url and the map url. YMMV.

Unless the current setup if for the development only, than my comment is moot and we can merge as-is.

File: resources/schema.edn
Comment (jpmonettas) [reply resources/schema.edn:69]: Yeah exactly. I feel we are repeating some common analysis things and just adding this notes to remember and think about it.

File: src/clj/api/resolvers.clj
Comment (jpmonettas) [reply src/clj/api/resolvers.clj:127]: This I'll take a look at what it takes to grab that bounding box from the analysis data and if it is super easy we can go with that, without adding it to the db

File: src/clj/api/server.clj
Comment (jpmonettas) [reply src/clj/api/server.clj:166]: Yeah this is not about reading the jar name. Is about reading the contents of a folder inside your jar resources. I tried a bunch of things but couldn't make it work. It should be possible but didn't want to spend more time on that since is nice to have but not a lot of value now.

File: src/cljc/shared/math_utils.cljc
Comment (jpmonettas) [reply src/cljc/shared/math_utils.cljc]: Damn, I'll change it everywhere XD

File: src/cljs/ui/component/maps.cljs
Comment (jpmonettas) [reply src/cljs/ui/component/maps.cljs:202]: Not needed

File: src/cljs/ui/component/maps.cljs
Comment (jpmonettas) [reply src/cljs/ui/component/maps.cljs:212]: Hahah poor parens

File: src/cljs/ui/config.cljs
Comment (jpmonettas) [reply src/cljs/ui/config.cljs:22]: Yeah all that is just for development purposes, just so we can test the technology

**Comment by jpmonettas** [comment]
Yeah all this is just to get the map stuff merged into master so we can move from there.
> my major one is that I imagined the renderer to be a separate client application (separate cljs build), running in it's own process w-out even calling graphQL

I didn't knew that but is completely possible, and probably a good idea also.
I can decouple that after the merge if you want, or I can also do it now before merge it, same for me.

File: src/clj/api/server.clj
Comment (fbielejec) [reply src/clj/api/server.clj:166]: I wasn't clear in my comment: the Java code below that line does exactly that, it unpacks the content of a jar. Look at the java.util.jar.JarEntry API

File: src/cljs/ui/component/maps.cljs
Comment (fbielejec) [reply src/cljs/ui/component/maps.cljs:202]: there's couple of places I saw this, but I marked it here only

**Review by fbielejec — APPROVED** [review]

**Pull request merged**
